source/4_tools/console_app/src/console_app.cpp

```diff
@@ -0,0 +1,109 @@
+#include "rex_engine/defines.h"
+#include "rex_engine/diagnostics/logging/log_macros.h"
+#include "rex_engine/entrypoint.h"
+#include "rex_engine/memory/memory_tracking.h"
+#include "rex_std/iostream.h"
+#include "rex_std_extra/utility/type_id.h"
+#include "rex_windows/console_application.h"
+#include "rex_engine/event_system.h"
+#include "rex_engine/filesystem/vfs.h"
+#include "rex_windows/input/input.h"
+
+#include <Windows.h>
+
+DEFINE_LOG_CATEGORY(LogConsoleApp, rex::LogVerbosity::Log);
+
+namespace rex
+{
+  void pointer_tracking_test()
+  {
+    // the following pointer should get tracked and automatically be processed by the tracking system
+    // depending on which one is enabled
+
+    // the code will follow the following step:
+    // operator new -> Global Allocator (Tracked) -> Global Allocator (Untracked) -> System Call
+    //                                                                                    |
+    //                                                                                    | 
+    // operator new <- Global Allocator (Tracked) <- Global Allocator (Untracked)  <------+ 
+    //                         |
+    //                         |
+    //                         v
+    //                   Save to file
+    // 
+    // Functionality should be implemented in memory_tracking.cpp
+    // However, we'll need a file system first to save things to disk
+    // We can then write a tool that scans this file for a memory analysis
+    // 
+    // Requirements of this tool:
+    // - Open a memory report file saved by the engine (See above)
+    // - Track all live allocations with their callstacks
+    // - Diff between different points in time
+    // - Track all single frame allocations
```

Comment:
this explanation should not go into the repo, just make a ticket for it in the project overview of rex

Reply:
the console app is not yet added to rex, I'm currently using it as a test app, so I don't think the above is a concern

Reply:
it's not added to rex solution, just to the repo